parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -35,31 +38,79 @@ cleanupContainerdDlFiles() {
     rm -rf $CONTAINERD_DOWNLOADS_DIR
 }
 
-installContainerRuntime() {
-    if [ "${NEEDS_CONTAINERD}" == "true" ]; then
-        echo "in installContainerRuntime - KUBERNETES_VERSION = ${KUBERNETES_VERSION}"
-        local containerd_version
-        if [ -f "$MANIFEST_FILEPATH" ]; then
-            containerd_version="$(jq -r .containerd.edge "$MANIFEST_FILEPATH")"
-            if [ "${UBUNTU_RELEASE}" == "18.04" ]; then
-                containerd_version="$(jq -r '.containerd.pinned."1804"' "$MANIFEST_FILEPATH")"
-            fi
-        else
-            echo "WARNING: containerd version not found in manifest, defaulting to hardcoded."
-        fi
+# After the centralized packages changes, the containerd versions are only available in the components.json.
+installContainerdWithComponentsJson() {
+    os=${UBUNTU_OS_NAME}
```

Comment:
could we factor this out to a somewhat generic `installComponent` function? seems a lot of this logic will be common across all components we install via the components file

Reply:
Hmm. There are 2 reasons I didn't do that. But I am open to that.
1. (Minor) This PR has been quite large. Refactoring the download/install logic will introduce another aspect of complexity which will make it even harder to review.
2. (Major) The download/install logic of each package is quite different. I left a comment in the code that
```*)
      echo "Package name: ${name} not supported for download. Please implement the download logic in the script."
      # We can add a common function to download a generic package here.
      # However, installation could be different for different packages.
      ;;
```

For the 5 packages we have now, we are already using various methods to download/install.
1. cri-tools: download using curl and install after artifact streaming. Not sure if it has to install after that.
2. azure-cni: download a tar ball using curl and untar
3. cni-plugins: similar to azure-cni
4. runc: For Ubuntu, it downloads a deb file using curl and install the deb file. For Mariner, it's included in the base image so do nothing.
5. containerd:
- Ubuntu: download/install either through url or apt-get
- Mariner: using dnf to install

We can still have a generic installComponent. But inside it will still check what package it is to use the corresponding method to download/install.